src/api/go/pxapi/opts.go

```diff
@@ -28,6 +34,21 @@ func WithCloudAddr(cloudAddr string) ClientOption {
 	}
 }
 
+// Allows disabling TLS verification if the `PX_DISABLE_TLS` env var is set and the cloud address is a cluster domain (cluster.local).
+func WithDisableTLSVerification(cloudAddr string) ClientOption {
+	return func(c *Client) {
+		isInternal := strings.Contains(cloudAddr, "cluster.local")
```

Comment:
nit: you can get the cloud addr from the client struct instead of passing it in

Reply:
That was my initial approach, but then the `ClientOption` has different behavior depending on the ordering. My unit tests passed but when I tried the example test it failed because I added the `WithDisableTLSVerification` one first (before the `Client` struct was mutated by `WithCloudAddr`).

I'm happy to change it back if you still think that is better, but I think we need to make the godoc string more clear on how to correctly use it then.

Reply:
Yeah I forgot the `cloudAddr` was a `ClientOption` too. This is fine.